### PR TITLE
mobile: Publish Android library (AAR) with xDS support

### PR DESCRIPTION
**Pull request by fredyw**
This PR builds a separate Android library (AAR) that supports xDS and publishes it to Maven Central. This library will have `artifactId`: `envoy-xds`.

Risk Level: low
Testing: CI
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Review by abeyad — APPROVED** [review]
awesome!

File: mobile/library/proguard.txt [hunk 1]
```diff
@@ -2,10 +2,12 @@
 -dontwarn kotlin.**
 -dontwarn org.jetbrains.annotations.NotNull
 -dontwarn org.jetbrains.annotations.Nullable
+-dontwarn com.google.protobuf.**
```
Comment (abeyad) [inline mobile/library/proguard.txt:5]: just curious, why were these changes needed?

**Comment by abeyad** [comment]
Oh nevermind, they were converted to non-required targets which is good

File: mobile/library/proguard.txt (hunk 1 above)
Comment (fredyw) [reply mobile/library/proguard.txt:5]: That caused the proguard build to fail.
